plugins/newsdownloader.koplugin/internaldownloadbackend.lua

```diff
@@ -0,0 +1,47 @@
+local http = require("socket.http")
+local https = require("ssl.https")
+local logger = require("logger")
+local ltn12 = require("ltn12")
+local socket = require('socket')
+local socket_url = require("socket.url")
+
+local InternalDownloadBackend = {}
+local max_redirects = 10;
+
+function InternalDownloadBackend:getResponseAsString(url, redirectCount)
+    if not redirectCount then
+        redirectCount = 0
+    elseif redirectCount == max_redirects then
+        logger.warn("InternalDownloadBackend: reached max redirects: ", redirectCount)
+    end
+    logger.dbg("InternalDownloadBackend: url :", url)
+    local request, sink = {}, {}
+    request['sink'] = ltn12.sink.table(sink)
+    request['url'] = url
+    local parsed = socket_url.parse(url)
+
+    local httpRequest = parsed.scheme == 'http' and http.request or https.request
+    -- first argument returned by skip is code
+    local _, headers, status = socket.skip(1, httpRequest(request))
```

Comment:
https://github.com/koreader/koreader/pull/3618#discussion_r164571382

> I don't have number (code only) status:(

This line is where you're explicitly ignoring it. 

```lua
local code, headers, status

Reply:
Thanks:)